### DISCUSSION TITLE
Ignore empty files and folders?

### DISCUSSION THREAD

**Discussion by ghost** [Dynatrace/dynatrace-configuration-as-code#216] [Archive]
Should monaco ignore empty files and folders? At present it errors if the JSON files are empty.

This is relevant to the new `monaco init` functionality which generates sample YAML and empty JSON files. As currently implemented, a user would have to delete any folders they're not using. Perhaps that's cleaner though as it forces a clean git repo and an understanding of the tool?

Relates to #128 

Discuss :)

**Comment by patrickpichler** [comment]
If you ask me, we should definitely not ignore empty files. 
I personally prefer monaco to be strict here, as empty files do not serve any purpose
and are kinda of a "config smell". 

As for `monaco init` it would probably be best to allow users to choose what configurations they
want a default for. Another feature which would maybe help improve the situation during initial deployment
would be to allow the deployment of only a set of configurations, similar to the way we can deploy only
certain projects yet. Then we can init with empty files and only deploy the config we are currently working on.

**Comment by ghost** [comment]
Commit `4f94274` adds the `-c` or `--createSpecificAPI` flag to the `monaco init` command which will then only create dummy folders for those APIs

```
monaco.exe init --createSpecificAPI auto-tag,dashboard
```

**Comment by patrickpichler** [comment]
That is awesome :+1:
